### PR TITLE
Prep for platform specific flow

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] move to haste
- [x] dont alias files
- [x] mobile files all .native
- [x] desktop files all .desktop
- [x] filter out most of node_modules. The regexp in flowconfig [ignore] section sucks but this makes my flow go from 30s -> 25s

This is a pr to break up into steps what this pr was trying to accomplish: https://github.com/keybase/client/pull/11907/files

### CODE REVIEW

**Comment by chrisnojima** [comment]
- [ ] desktop test build

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is a partial step towards doing per platform flow checking

**Comment by adamjspooner** [comment]
Cool. Thanks for the explanation!

File: shared/.flowconfig [hunk 1]
```diff
@@ -25,9 +39,23 @@ esproposal.class_static_fields=enable
...
+module.system=haste
+
+module.system.haste.use_name_reducers=true
+# strip .js or .js.flow suffix
+module.system.haste.name_reducers='^\(.*\)\.js\(\.flow\)?$' -> '\1'
+# strip .mobile suffix
+# module.system.haste.name_reducers='^\(.*\)\.ios$' -> '\1'
+# module.system.haste.name_reducers='^\(.*\)\.android$' -> '\1'
+# module.system.haste.name_reducers='^\(.*\)\.native$' -> '\1'
+# module.system.haste.name_reducers='^\(.*\)\.desktop\.js$' -> '\1'
```
Comment (buoyad) [inline shared/.flowconfig]: should this include the `.js`?

Comment (chrisnojima) [reply shared/.flowconfig]: i think the other mapper runs first actually. anyways its commented out for a later phase. we'll figure it out then (this came from RN's .flowconfig actually)

**Pull request merged**